### DISCUSSION TITLE
How to implement an inlineCompletionItem extension ?

### DISCUSSION THREAD

**Discussion by znoliver** [eclipse-theia/theia#13473] [Q&A] [answered]
I want to develop a extension to implement AI code hint ，I plan to access OpenAI by HTTP API, but how to display the result in editor? I find vscode api has InlineCompletionItemProvider , what about Theia?

Example ：
[Example](https://exafunction.github.io/public/autocomplete/multi_line.gif)

**Comment by znoliver** [comment]
Can someone help me？

**Comment by msujew** [comment, marked as answer]
Hey @znoliver,

the recommended solution would be to create a vscode extension that contributes this feature. That way, you don't have to work with Theia internal code (which can regularly change). Alternatively, here's how the vscode API is contributed on the browser level of Theia:

https://github.com/eclipse-theia/theia/blob/9ff0cedff1d591b0eb4be97a05f6d992789d0a24/packages/plugin-ext/src/main/browser/languages-main.ts#L923-L937

The goal would be to do something similar to accomplish what you're trying to do.

**Reply by znoliver** [reply]
Thanks
